cpp/benchmarks/copying/contiguous_split_benchmark.cu

```diff
@@ -14,21 +14,28 @@
  * limitations under the License.
  */
 
-#include <benchmark/benchmark.h>
+#include <benchmarks/fixture/benchmark_fixture.hpp>
+#include <benchmarks/synchronization/synchronization.hpp>
 
-#include <cudf/column/column.hpp>
+#include <cudf_test/column_wrapper.hpp>
 
+#include <cudf/column/column.hpp>
 #include <cudf/copying.hpp>
 
-#include <benchmarks/fixture/benchmark_fixture.hpp>
-#include <benchmarks/synchronization/synchronization.hpp>
-#include <cudf_test/column_wrapper.hpp>
+#include <thrust/iterator/detail/tagged_iterator.h>
+#include <thrust/random.h>
 
 // to enable, run cmake with -DBUILD_BENCHMARKS=ON
 
+template <typename Tag, typename Iterator>
+inline auto make_tagged_iterator(Iterator iter)
```

Comment:
This method, or the location it is called, need comments to explain why we need to add the iterator tag.

I also noticed that this tag is always added to cudf's counting-transform iterators. Is there a better way to leverage Thrust to make a device-tagged counting transform iterator? It would be nice to avoid calling `detail` methods from Thrust if there's another way to do this. (I'm not sure how a "device" tag is added to an iterator without this `detail` method, but I assume it has to do with how the iterator is constructed.)